src/mongocxx/test/validation_criteria.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "catch.hpp"
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/document/element.hpp>
+
+#include <mongocxx/validation_criteria.hpp>
+
+using namespace mongocxx;
+using namespace bsoncxx;
+
+TEST_CASE("validation_criteria", "[validation_criteria]") {
+    validation_criteria criteria;
+
+    auto doc = builder::stream::document{} << "email" << builder::stream::open_document << "$exists"
+                                           << "true" << builder::stream::close_document
+                                           << builder::stream::finalize;
```

Comment:
Do we need to call finalize? If so, it's definitely wrong in other tests / examples.